downstairs/src/http.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2021 Oxide Computer Company
+use super::*;
+
+use dropshot::{
+    endpoint, ApiDescription, ConfigDropshot, HttpError, HttpResponseCreated,
+    HttpServerStarter, Path, RequestContext, TypedBody,
+};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+pub struct ServerContext {
+    downstairs: Mutex<HashMap<Uuid, Arc<Mutex<Downstairs>>>>,
+}
+
+#[derive(Deserialize, JsonSchema)]
+pub struct RunDownstairsForRegionParams {
```

Comment:
This is really just a copy of whatever options are in `downstairs/src/main.rs` for the `enum Args::Run` right?  I assume there is no easy way to re-use that, otherwise you would have.

Reply:
It is, and I prefer to have a separate type for it :)

Reply:
Keeping the two apart just means someone (you and I probably) need to remember to update the fields in the other struct as well.

Reply:
Resolved!